skbio/stats/composition.py

```diff
@@ -608,10 +611,272 @@ def centralize(mat):
+    Also casts composition into a pandas dataframe
+
+    Parameters
+    ----------
+    x : array_like or pd.DataFrame
+       Input composition matrix
+       where rows=samples and columns=features
+    ignore_zeros : bool
+       Specifies if compositions are allowed to have zeros
+
+    Returns
+    -------
+    pd.DataFrame
+       Validated composition matrix
+    """
+    if x.ndim > 2:
+        raise ValueError("Input matrix can only have two dimensions or less")
+    if isinstance(x, pd.DataFrame):
+        mat = closure(x)
+        samp_labs = x.index
+        feat_labs = x.columns
+    else:
+        mat = np.atleast_2d(x)
+        r, c = mat.shape
+        samp_labs = range(r)
+        feat_labs = range(c)
+    if np.any(mat == 0) and not ignore_zeros:
+        raise ValueError('Cannot handle zeros in compositions. '
+                         'Make sure to run a zero replacement method')
+    return pd.DataFrame(mat, index=samp_labs, columns=feat_labs)
+
+
+def _holm_bonferroni(p):
+    """
+    Performs Holm-Bonferroni correction for pvalues
+    to account for multiple comparisons
+
+    Parameters
+    ---------
+    p: numpy.array
+        array of pvalues
+
+    Returns
+    -------
+    numpy.array
+        corrected pvalues
+    """
+    K = len(p)
+    sort_index = -np.ones(K, dtype=np.int64)
+    sorted_p = np.sort(p)
+    sorted_p_adj = sorted_p*(K-np.arange(K))
+    for j in range(K):
+        idx = (p == sorted_p[j]) & (sort_index < 0)
+        num_ties = len(sort_index[idx])
+        sort_index[idx] = np.arange(j, (j+num_ties), dtype=np.int64)
+
+    sorted_holm_p = [min([max(sorted_p_adj[:k]), 1])
+                     for k in range(1, K+1)]
+    holm_p = [sorted_holm_p[sort_index[k]] for k in range(K)]
+    return holm_p
+
+
+def _log_compare(mat, cats,
+                 significance_test=scipy.stats.ttest_ind):
+    """
+    Calculates pairwise log ratios between all otus and performs a permutation
+    tests to determine if there is a significant difference in OTU ratios with
+    respect to the variable of interest
+
+    Parameters
+    ----------
+    mat: np.array
+       rows = samples
+       columns = features (i.e. OTUs)
+    cats: np.array, float
+       Vector of categories
+    significance_test: function
+        statistical test to run
+
+    Returns:
+    --------
+    log ratio pvalue matrix
+    """
+    r, c = mat.shape
+    log_ratio = np.zeros((c, c))
+    log_mat = np.log(mat)
+    cs = np.unique(cats)
+
+    def func(x):
+        return significance_test(*[x[cats == k] for k in cs])
+
+    for i in range(c-1):
+        ratio = (log_mat[:, i].T - log_mat[:, i+1:].T).T
+        m, p = np.apply_along_axis(func,
+                                   axis=0,
+                                   arr=ratio)
+        log_ratio[i, i+1:] = np.squeeze(np.array(p.T))
+    return log_ratio
+
+
 def _gram_schmidt_basis(n):
     """
     Builds clr transformed basis derived from
```

Comment:
This isn't formatted correctly (the other private functions in `composition` aren't either). The docstring summary line should match the guidelines [here](https://github.com/numpy/numpy/blob/master/doc/HOWTO_DOCUMENT.rst.txt#sections) (see "1. Short summary" section).

It isn't a big deal because these functions are private. Often times we aren't even enforcing numpydoc on private functionality, just filling in as much detail as necessary for maintainers to grok. Up to you if you want to fix these private function docstrings, I won't be reviewing them in any detail.